File: check_translations.py

import re

file_path = '/home/kira/Desktop/desarrolloia/.obsidian/plugins/finance-os-plugin2/src/utils/translations.ts'

with open(file_path, 'r') as f:
    content = f.read()

# simpler regex for Python
en_match = re.search(r"en: \{([\s\S]*?)\n\s*\},", content)
es_match = re.search(r"es: \{([\s\S]*?)\n\s*\},", content) # Adjusted regex for potentially closing braces

def parse_keys(text):
    keys = set()
    for line in text.splitlines():
        match = re.search(r"^\s*'([\w\.]+)'\s*:", line)
        if match:
            keys.add(match.group(1))
    return keys

if en_match and es_match:
    en_keys = parse_keys(en_match.group(1))
    es_keys = parse_keys(es_match.group(1))
    
    missing_in_es = list(en_keys - es_keys)
    missing_in_en = list(es_keys - en_keys)
    
    print("Missing in ES:", missing_in_es)
    print("Missing in EN:", missing_in_en)
else:
    print("Could not find en or es objects")

File: check_translations.ts

import * as fs from 'fs';
import * as path from 'path';

// Read the file
const content = fs.readFileSync('/home/kira/Desktop/desarrolloia/.obsidian/plugins/finance-os-plugin2/src/utils/translations.ts', 'utf8');

// Extract the en and es objects
// refined regex to capture the object content
const enMatch = content.match(/en: \{([\s\S]*?)\n\s*\},/);
const esMatch = content.match(/es: \{([\s\S]*?)\n\s*\},?/);

if (!enMatch || !esMatch) {
    console.error("Could not find en or es objects");
    process.exit(1);
}

const parseKeys = (str: string) => {
    const keys = new Set<string>();
    const lines = str.split('\n');
    for (const line of lines) {
        const match = line.match(/^\s*'([\w.]+)':/);
        if (match) {
            keys.add(match[1]);
        }
    }
    return keys;
};

const enKeys = parseKeys(enMatch[1]);
const esKeys = parseKeys(esMatch[1]);

const missingInEs = [...enKeys].filter(k => !esKeys.has(k));
const missingInEn = [...esKeys].filter(k => !enKeys.has(k));

console.log("Missing in ES:", missingInEs);
console.log("Missing in EN:", missingInEn);
